packages/Meshfree/src/DTK_MovingLeastSquaresOperator_def.hpp

```diff
@@ -0,0 +1,133 @@
+/****************************************************************************
+ * Copyright (c) 2012-2017 by the DataTransferKit authors                   *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the DataTransferKit library. DataTransferKit is     *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ ****************************************************************************/
+
+#ifndef DTK_MOVING_LEAST_SQUARES_OPERATOR_DEF_HPP
+#define DTK_MOVING_LEAST_SQUARES_OPERATOR_DEF_HPP
+
+#include <DTK_DBC.hpp>
+#include <DTK_DetailsMovingLeastSquaresOperatorImpl.hpp>
+#include <DTK_DetailsNearestNeighborOperatorImpl.hpp> // makeDistributedSearchTree, fetch
+#include <DTK_DistributedSearchTree.hpp>
+
+namespace DataTransferKit
+{
+
+template <typename DeviceType, typename CompactlySupportedRadialBasisFunction,
+          typename PolynomialBasis>
+MovingLeastSquaresOperator<DeviceType, CompactlySupportedRadialBasisFunction,
+                           PolynomialBasis>::
+    MovingLeastSquaresOperator(
+        Teuchos::RCP<Teuchos::Comm<int> const> const &comm,
+        Kokkos::View<Coordinate const **, DeviceType> source_points,
+        Kokkos::View<Coordinate const **, DeviceType> target_points,
+        Teuchos::ParameterList const &plist )
+    : _comm( comm )
+    , _offset( "offset" )
+    , _ranks( "ranks" )
+    , _indices( "indices" )
+    , _coeffs( "polynomial_coefficients" )
+{
+    double const radius = plist.get<double>( "radius" );
```

Comment:
Note that we are looking at getting rid of the radius as an user input.  We would go for a kNN search instead which means we would have variable radii for different target points.